src/app/services/app-auth.service.ts

```diff
@@ -13,8 +14,13 @@ import { BehaviorSubject } from 'rxjs';
   providedIn: 'root',
 })
 export class AppAuthService {
-  private isAuthenticated$$ = new BehaviorSubject<boolean>(false);
-  public isAuthenticated$ = this.isAuthenticated$$.asObservable();
+  public isAuthenticated$ = this.oauthService.events.pipe(
+    map(() => this.oauthService.hasValidAccessToken()),
+    shareReplay({
```

Comment:
@kirillsud and @sergii-backbase I was forced to add shareReplay to make it stay alive. Let me know if you agree